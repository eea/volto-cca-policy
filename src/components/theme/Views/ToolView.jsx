import React from 'react';
import { HTMLField, ContentMetadata } from '@eeacms/volto-cca-policy/helpers';

function ToolView(props) {
  const { content } = props;

  console.log(content);

  return (
    <div className="tool-view">
      <div style={{}}>
        <ContentMetadata {...props} />
      </div>
      <h1>{content.title}</h1>
      <h4>Description</h4>
      <HTMLField
        value={content.long_description}
        className="long_description"
      />
      <hr />
      <h4>Reference information</h4>
      <h5>Websites</h5>
      {content.websites.map((url) => (
        <a key={url} href={url}>
          {url}
        </a>
      ))}
      <h5>Source</h5>
      <HTMLField value={content.source} />
    </div>
  );
}

export default ToolView;
