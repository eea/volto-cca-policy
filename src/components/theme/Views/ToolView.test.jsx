import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-intl-redux';
import ToolView from './ToolView';
import renderer from 'react-test-renderer';

const mockStore = configureStore();

jest.mock('semantic-ui-react', () => ({
  ...jest.requireActual('semantic-ui-react'),
}));

describe('ToolView', () => {
  it('should render the component', () => {
    const content = {
      title: 'My ToolView',
      geochars:
        '{\r\n "geoElements":{"element":"GLOBAL",\r\n "macrotrans":null,"biotrans":null,"countries":[],\r\n "subnational":[],"city":""}}',
    };

    const store = mockStore({
      userSession: { token: '1234' },
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <ToolView content={content} />
        </MemoryRouter>
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
