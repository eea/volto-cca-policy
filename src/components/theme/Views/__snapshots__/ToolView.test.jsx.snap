// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolView should render the component 1`] = `
<div
  className="tool-view"
>
  <div
    className="ui container"
  >
    <div
      className="ui twelve column grid"
    >
      <div
        className="row"
      >
        <div
          className="nine wide computer twelve wide mobile twelve wide tablet column col-left"
        >
          <div
            className="ui label"
          >
            Tools
          </div>
          <h1>
            My ToolView
          </h1>
          <h4>
            Description
          </h4>
          <div
            className="long_description"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
          <hr />
          <h4>
            Reference information
          </h4>
          <div
            className="published-modified-info"
          >
            <p>
              <span>
                <strong>
                  Published in Climate-ADAPT
                </strong>
                 
                Invalid Date
              </span>
              <span>
                   -   
              </span>
              <span>
                <strong>
                  Last Modified in Climate-ADAPT
                </strong>
                 
                Invalid Date
              </span>
            </p>
          </div>
        </div>
        <div
          className="three wide computer twelve wide mobile twelve wide tablet column col-right"
        >
          <div
            style={Object {}}
          >
            <div
              className="content-metadata"
            >
              <h5>
                Geographic characterisation:
              </h5>
              <div
                className="geochar"
              >
                <p>
                  Global
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
